app/src/projects.js

```diff
@@ -54,9 +54,42 @@ router.use(auth.middleware.credentials);
 /**
  * Show a list of all the users projects.
  */
-router.get('/', (req, res) => {
-    db.get().collection('projects').find({ "owner": req.cookies.u, $or: [{ hidden: { $exists: false } }, { hidden: false }] }).project({ title: true, id: true }).toArray((err, projects) => {
-        ejs.renderFile(`app/views/projects.ejs`, { projects_data: projects }, {}, (err, str) => {
+router.get('/', async (req, res) => {
+
+    // Create array of last modified timeStamps for each project
+
+    let timeStamps = [];
+    let timeMap = {};
+    findMTime = db.get().collection('project_data').find({});
+    await findMTime.forEach(
+        doc => {
+            timeStamps.push(doc._m.mtime);
+            let forConversion = new Date(doc._m.mtime).toString();
+            let splitter = forConversion.split("T");
+            // console.log("Last edited on: " + splitter[0]);
+            timeMap[doc._id] = splitter[0];
+        }
+    );
+    // console.log(timeStamps)
+
+    // Create a map for {project UID : last modification timeStamp} , filtering on previously
+    // created array
+    let editMap = {};
+    lastEdits = db.get().collection('o_project_data').find({ "m.ts": { $in: timeStamps } });
+    await lastEdits.forEach(
+        doc => {
+            // console.log(doc);
+            editMap[doc.d] = doc.m.userid;
+        }
+    );
```

Comment:
This is also going to get every operational document from the `o_project_data` collection which is expected to be a **massive** database since every character stroke is stored.

Reply:
For reference, my `o_project_data` has 2,800 entries which have all only been generated over the past few days while working on this.